src/mpi/romio/adio/common/ad_fstype.c

```diff
@@ -370,6 +386,23 @@ static int romio_statfs(const char *filename, int64_t * file_id)
         }
     }
 #endif
+    /* fuse is a special case: normally we don't care if a file is on fuse
+     * or not, unless that file is actually a DAOS file */
+    if (*file_id == FUSE_SUPER_MAGIC) {
+        *file_id = UNKNOWN_SUPER_MAGIC;
+#ifdef ROMIO_DAOS
+        int ret;
+        /* DAOS adds a wrinkle here:  fuse overwrites the backend's stat structure
+         * with the fuse filesystem type.  For DAOS, we have to make a special call
+         * to see if a file on FUSE is actually DAOS */
+        struct duns_attr_t attr = { 0 };;
+
+        ret = duns_resolve_path(filename, &attr);
```

Comment:
im not too familiar with the internals here of romio_statfs. so some questions:
1) if several ranks open the file collectively, does this end up being called on every rank?
2) if one specified daos: prefix or sets the env variable `ROMIO_FSTYPE_FORCE="daos:"`, is this still called?

I did some local testing and i came to the conclusion that the answers are:
1) no
2) no
which is great, but just wanted to verify

Reply:
yeah, check out `ADIO_FileSysType_fncall_scalable` which we can call unless NFS is enabled 

(NFS?  yeah... imagine a system where a file system lives on rank 0 or the "master" node and the other nodes in the system NFS mount it..  well it was a bug report back in 2007 i am so old...)